### ISSUE TITLE
Discord in ru rippppppp

### ISSUE THREAD

**Issue by idontliveincrimea** [MatsuriDayo/nekoray#1448]
For some reason, when using nekoray vless, it does not help with bypassing the discord voice block in Russia

**Comment by Akiyamov** [comment]
Добавь приложение дискорда в процессы для обхода в кастомных маршрутах. Для загрузки видео нужно добавить discordapp.com потому что он открывает браузер для загрузки. Пример.
---
Add discord process in custom routes. To download videos add discordapp.com since discord app opens browser to download video. An example.
```json
{
    "rules": [
        {
            "outbound": "dns-out",
            "protocol": "dns"
        },
        {
            "outbound": "block",
            "protocol": [
                "quic"
            ]
        },
        {
            "geoip": "antizapret",
            "geosite": "antizapret",
            "outbound": "proxy"
        },
        {
            "domain_suffix": [
                ".ggpht.com",
                ".googlevideo.com",
                ".youtube.com",
                ".ytimg.com",
                ".googleapis.com",
                "discordapp.com"
            ],
            "outbound": "proxy"
        },
        {
            "outbound": "proxy",
            "process_name": [
                "Discord.exe"
            ]
        }
    ]
}

```

**Comment by ag1lebyte** [comment]
@Akiyamov, доброго времени суток. Подскажите куда это вставлять, а то я только вчера установил програмку и купил проксю, до этого никогда с ними не работал. Заранее огромное спасибо

**Comment by Akiyamov** [comment]
В папку с исполняемым файлом nekobox'а закидываются эти файлы
https://github.com/savely-krasovsky/antizapret-sing-box/releases/latest/download/geoip.db
https://github.com/savely-krasovsky/antizapret-sing-box/releases/latest/download/geosite.db
Текст выше в Настройки -> Настройки маршрутов -> третья вкладка сверху -> кастомные маршруты. В третьей вкладке также поменять стандартный outbound на bypass

**Comment by indeboss** [comment]
> В папку с исполняемым файлом nekobox'а закидываются эти файлы https://github.com/savely-krasovsky/antizapret-sing-box/releases/latest/download/geoip.db https://github.com/savely-krasovsky/antizapret-sing-box/releases/latest/download/geosite.db Текст выше в Настройки -> Настройки маршрутов -> третья вкладка сверху -> кастомные маршруты. В третьей вкладке также поменять стандартный outbound на bypass

Появляется такая ошибка: 
![image](https://github.com/user-attachments/assets/cf91e27b-b0a7-4819-8de4-5967e5dcb016)
Тип подключения - shadowsocks

**Comment by Akiyamov** [comment]
Ядро sing-box надо использовать. Основные настройки -> Ядро и там выбираете sing-box

**Comment by Goodron2** [comment]
For some reason, custom json doesn't work for me too. So I have to configure routing.

here is the list of discord IPs  https://gitlab.com/GhostRooter0953/discord-voice-ips/-/blob/main/discord-voice-ip-list?ref_type=heads

and also use `geosite:discord` 
![image](https://github.com/user-attachments/assets/8e9e7681-d0f3-4fdb-b026-54677959746d)

**Comment by Akiyamov** [comment]
Don't input all ips like this. You can use `ip_cidr`. I use sing-box without nekobox but it shouldn't be any different. Here are cidrs for all discord servers: 
- 66.22.0.0/16 
- 34.0.0.0/16 
- 35.207.0.0/16 
- 35.214.0.0/16 
- 35.217.0.0/16 
- 35.215.0.0/16 
- 35.213.0.0/16 
- 35.219.0.0/16 
- 5.200.14.249 
Also it should be possible to just use regex since link to voice servers starts with `wss://` AND it should be possible to route all ports from 50000-65535 since they are used by discord

**Comment by VapeNationGitHub** [comment]
> В папку с исполняемым файлом nekobox'а закидываются эти файлы https://github.com/savely-krasovsky/antizapret-sing-box/releases/latest/download/geoip.db https://github.com/savely-krasovsky/antizapret-sing-box/releases/latest/download/geosite.db Текст выше в Настройки -> Настройки маршрутов -> третья вкладка сверху -> кастомные маршруты. В третьей вкладке также поменять стандартный outbound на bypass

Аналогичная ошибка:
![Аннотация 2024-10-12 223406](https://github.com/user-attachments/assets/9cab31a5-b2d7-4b47-b5e3-fd3bcffcf2f1)

**Comment by Akiyamov** [comment]
Я же ответил на то сообщение, что надо ядро sing-box

**Comment by 17kato** [comment]
> Я же ответил на то сообщение, что надо ядро sing-box

привет, есть способ как-то с тобой связаться? у меня есть ошибка, и я смотрю ты понимаешь, что и как в этой программе делать. Я облазил весь интернет, так и не нашел ответа

**Comment by zomba444ok** [comment]
@17kato писать о своей проблеме сразу во всех issue не принято. Одной темы достаточно. Если кто-то захочет/сможет помочь, он поможет

**Comment by ag1lebyte** [comment]
> Don't input all ips like this. You can use `ip_cidr`. I use sing-box without nekobox but it shouldn't be any different. Here are cidrs for all discord servers:
> 
>     * 66.22.0.0/16
> 
>     * 34.0.0.0/16
> 
>     * 35.207.0.0/16
> 
>     * 35.214.0.0/16
> 
>     * 35.217.0.0/16
> 
>     * 35.215.0.0/16
> 
>     * 35.213.0.0/16
> 
>     * 35.219.0.0/16
> 
>     * 5.200.14.249
>       Also it should be possible to just use regex since link to voice servers starts with `wss://` AND it should be possible to route all ports from 50000-65535 since they are used by discord

Не подскажешь как вот эти ip cidr получить можно? Может быть есть сайты, где домен ввел и тебе высветилось или это все ручками надо делать с использованием чего-то? Просто на будущее, вдруг еще что-то кроме дс отлетит

**Comment by equuskk** [comment]
> > Don't input all ips like this. You can use `ip_cidr`. I use sing-box without nekobox but it shouldn't be any different. Here are cidrs for all discord servers:
> > ```
> > * 66.22.0.0/16
> > 
> > * 34.0.0.0/16
> > 
> > * 35.207.0.0/16
> > 
> > * 35.214.0.0/16
> > 
> > * 35.217.0.0/16
> > 
> > * 35.215.0.0/16
> > 
> > * 35.213.0.0/16
> > 
> > * 35.219.0.0/16
> > 
> > * 5.200.14.249
> >   Also it should be possible to just use regex since link to voice servers starts with `wss://` AND it should be possible to route all ports from 50000-65535 since they are used by discord
> > ```
> 
> Не подскажешь как вот эти ip cidr получить можно? Может быть есть сайты, где домен ввел и тебе высветилось или это все ручками надо делать с использованием чего-то? Просто на будущее, вдруг еще что-то кроме дс отлетит

https://github.com/rekryt/iplist/blob/master/config/discord/discord.gg.json

**Comment by Akiyamov** [comment]
@ag1lebyte вот с этого сайта https://iplist.opencck.org/
Пример того как он спарсил все зоны для голосовух https://iplist.opencck.org/?format=text&data=cidr4&site=discord.gg&site=discord.media

**Comment by Akiyamov** [comment]
Также еще есть репозиторий который хранит в себе json и srs для крупного списка сайтов. https://github.com/MetaCubeX/meta-rules-dat/tree/sing/geo
Если комьюнити добавит поддержку срс в некобокс то добавлять сайты можно будет в разы удобнее. На чистом синг боксе это делается без проблем

**Comment by Xixo1** [comment]
> Также еще есть репозиторий который хранит в себе json и srs для крупного списка сайтов. https://github.com/MetaCubeX/meta-rules-dat/tree/sing/geo Если комьюнити добавит поддержку срс в некобокс то добавлять сайты можно будет в разы удобнее. На чистом синг боксе это делается без проблем

То есть сейчас редактор JSON встроенный в некобокс не поддерживает загрузку списков через url? Почему-то думал что синг-бокс и редактор в некобоксе это одно и тоже...

**Comment by Akiyamov** [comment]
Нет, не одно и тоже. То, что сейчас в некобоксе это корректировка rules и только rules. Если бы это был routes, то проблем нет. Пример: https://github.com/MatsuriDayo/nekoray/issues/1446

**Comment by Xixo1** [comment]
> Нет, не одно и тоже. То, что сейчас в некобоксе это корректировка rules и только rules. Если бы это был routes, то проблем нет. Пример: https://github.com/MatsuriDayo/nekoray/issues/1446

Понял, спасибо. Сегодня пытался похожий код написать чтобы подтягивал списки и проксировал, но тоже ошибки и голова задымилась) На ios с этой задачей отлично справляется shadowrocket, странно что на пк еще не допилили такой функционал хотя бы в некобокс. Hiddify неплох дизайном, но настроек почти нет, минимализм. Кто-то писал, что можно в некобоксе дополнить свой конфиг настройками (белый список приложений и т.д.), затем выгрузить и заливать в другие клиенты, но чет хз, может ли так работать?

**Comment by Akiyamov** [comment]
Надо сделать выбор geoip/geosite дб в настройках просто и потом уже включать его руками либо сделать темплейт для рф. В телефоне такой функционал есть на некобоксе. Можно выгрузить конфиг, но лучше не стоит. Максимум вытащить outbound профиля и вставить в нормальный конфиг
